tools/azure-devtools/src/azure_devtools/perfstress_tests/_perf_stress_runner.py

```diff
@@ -215,7 +218,7 @@ async def start(self):
 
             # If a warm up is configured, wait till all tests have finished all setup
             # stages before beginning "Warmup".
-            if self.per_test_args.warmup:
+            if self.per_test_args.warmup and not self.per_test_args.profile:
```

Comment:
What was causing the deadlock here?

Reply:
In the `_start_tests` method, we skip the synchronization step for "Warmup" if [profiling is enabled](https://github.com/Azure/azure-sdk-for-python/blob/main/tools/azure-devtools/src/azure_devtools/perfstress_tests/_perf_stress_proc.py#L63). If we run `self._next_stage("Warmup"`, then don't run ` _synchronize(test_stages["Warmup"])`, the following `_synchronize(test_stages["Tests"])` gets stuck. So, here just making sure the conditional checks align.